### DISCUSSION TITLE
Request for CAD sketches of head-hand kinematic chain

### DISCUSSION THREAD

**Discussion by vicentepedro** [robotology/community#78] [Q&A] [answered]
Hi all!

I need the kinematic structure of iCub from the head to the right hand in order to show how the joints are placed in the robot.

I have found this picture of the head:
![headcadrefframe](https://cloud.githubusercontent.com/assets/9551355/11365175/91165182-929c-11e5-8a95-a89780abbd0d.jpg)

and this one for the right hand (with the reference frame attached)
![righthandcadrefframe](https://cloud.githubusercontent.com/assets/9551355/11365177/93a1ee8e-929c-11e5-9d95-bea4980e5775.jpg)

I was wondering if there is a picture with the rest of the right arm, from the neck to the wrist.

Thanks in advance,

Pedro Vicente

**Comment by pattacini** [comment]
Perhaps @super-ste and/or @robotology/icub-support-team can help you out.

**Comment by super-ste** [comment]
Hi,
did you try having a look here: http://wiki.icub.org/wiki/ICubForwardKinematics

Hopefully you will find the answer there (according to the version of the robot you consider).

**Comment by vicentepedro** [comment]
Yes.
I found there the two images shown in the previous post. The head and the right hand.

Unfortunately, I didn't find anything for the rest of the arm. Just some plots in matlab, not the CAD design.

**Comment by pattacini** [comment]
@vicentepedro the CAD drawings are stored in our `SVN` repository here: https://svn.robotology.eu/repos/iCubHardware-pub/trunk/mechanics/

I think you could open the right drawing with applications like **Creo** or **ProE**.

@super-ste can correct me if I'm wrong, and possibly point you at the proper file.

**Comment by super-ste** [comment]
While what @pattacini says is correct, you can also have a look at the instructions here on how to download the CAD files:
http://wiki.icub.org/wiki/RobotCub

Or, if you just need a screenshot or anything like that, can you elaborate more on what you actually need? Maybe it's just lacking and we can add it to the wiki...

**Comment by vicentepedro** [comment]
Thanks @pattacini and @super-ste. 

I will try to install **Creo** or **ProE** and open the CAD files. I will let you know if it worked well.

Actually, the CAD pictures of the arms are missing on the wiki, maybe it's a good idea to add it.

**Comment by super-ste** [comment, marked as answer]
Another suggestion, if you just need to visualize the robot, you can install the [CREO View Express](http://www.ptc.com/product/creo/visualization/view-mcad/extension/express)
(much lighter than CREO Parametric and for free).

The file you want to look for and open could be (after checking out the whole repository) is:

trunk/mechanics/public/icub1_icub2/icub.asm
or
trunk/mechanics/public/icub1_icub2/icub2.asm

or use any combination of the pics below

![rc_tlr_003_a_005_upperbody_front](https://cloud.githubusercontent.com/assets/8122012/11441836/2f3ab7ac-9510-11e5-9f8b-c37f1b31c0f2.jpg)
![rc_tlr_003_a_005_upperbody_top](https://cloud.githubusercontent.com/assets/8122012/11441835/2f395b28-9510-11e5-889e-fab503fb00b1.jpg)
![rc_tlr_003_a_005_upperbody_3d](https://cloud.githubusercontent.com/assets/8122012/11441837/2f3b482a-9510-11e5-9690-b00b34a55af3.jpg)

**Comment by vicentepedro** [comment]
Thanks, @super-ste.

The pics seem enough for my purpose. ;)
